### PR TITLE
[WIP] Basic Android QtService framework

### PR DESCRIPTION
**Pull request by nirvn**
This offers the skeleton of a basic functional Android QtService framework.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d19be7387b47d0d8a6eebb144ac73edc3e94522b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-a30bf905d627de4491b137eb66684a8522f086bf-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-a30bf905d627de4491b137eb66684a8522f086bf-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-a30bf905d627de4491b137eb66684a8522f086bf-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-a30bf905d627de4491b137eb66684a8522f086bf-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 87acf9a8717af1c92f72963cd78048cd064df54c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-10ef94d060175bd874d2035b68ade00190682459-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-10ef94d060175bd874d2035b68ade00190682459-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-10ef94d060175bd874d2035b68ade00190682459-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-10ef94d060175bd874d2035b68ade00190682459-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 37c9e0f52c26448157abcf8982b24e8ae7865f70:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-3d80271d46e2d3eafb93ee90e22fddce974b1f00-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-3d80271d46e2d3eafb93ee90e22fddce974b1f00-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-3d80271d46e2d3eafb93ee90e22fddce974b1f00-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-3d80271d46e2d3eafb93ee90e22fddce974b1f00-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 7cd3c10c69f774d7ba32685b165eed8a01ccdec8:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-17e195c89e5843f0af46f0b8d4b739fc413ee1af-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-17e195c89e5843f0af46f0b8d4b739fc413ee1af-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-17e195c89e5843f0af46f0b8d4b739fc413ee1af-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-17e195c89e5843f0af46f0b8d4b739fc413ee1af-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9d3d89c6e0c7a4149ab6abec4d5fefd0f06730bd:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-74e07bf5943e083c672dab91f4962481e1e92b91-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-74e07bf5943e083c672dab91f4962481e1e92b91-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-74e07bf5943e083c672dab91f4962481e1e92b91-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-74e07bf5943e083c672dab91f4962481e1e92b91-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 138ac8da9a7fd307afa5bc6159d9ff9b8ac8f623:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5ea0db1b6a31869e7a6f2f2681cb3442f13563e8-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5ea0db1b6a31869e7a6f2f2681cb3442f13563e8-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5ea0db1b6a31869e7a6f2f2681cb3442f13563e8-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5ea0db1b6a31869e7a6f2f2681cb3442f13563e8-x86.apk)

**Comment by nirvn** [comment]
Alright, this seems to have unlocked service launch:

`07-23 17:39:53.376  1254  1621 I ActivityManager: Start proc 2899:ch.opengis.qfield_beta:qt/u0a309 for service {ch.opengis.qfield_beta/ch.opengis.qfield_beta.QFieldService}`

We have our framework!

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 37d938046ea3ff67029b60e4d1e2e35e01b52275:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-395098141e7016c8891ddd271f23dac4a3516cba-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-395098141e7016c8891ddd271f23dac4a3516cba-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-395098141e7016c8891ddd271f23dac4a3516cba-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-395098141e7016c8891ddd271f23dac4a3516cba-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8345842b51423d121072ca28a665083b973083d6:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5c6d6351ee00649613581bd3a2b9d83a119c50ed-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5c6d6351ee00649613581bd3a2b9d83a119c50ed-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5c6d6351ee00649613581bd3a2b9d83a119c50ed-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2054-5c6d6351ee00649613581bd3a2b9d83a119c50ed-x86.apk)

File: scripts/docker-build.sh [hunk 1]
```diff
@@ -64,6 +64,7 @@ fi
 if [[ "X${APP_PACKAGE_NAME}" != "Xqfield" ]]; then
   grep "ch.opengis.qfield" -l -r ${SOURCE_DIR}/android/ | xargs sed -i "s/ch.opengis.qfield/ch.opengis.${APP_PACKAGE_NAME}/g"
   grep "ch\.opengis\.qfield" -l -r ${SOURCE_DIR}/src/ | xargs sed -i "s/ch\.opengis\.qfield/ch.opengis.${APP_PACKAGE_NAME}/g"
+  grep "ch\/opengis\/qfield" -l -r ${SOURCE_DIR}/src/ | xargs sed -i "s/ch\/opengis\/qfield/ch\/opengis\/${APP_PACKAGE_NAME}/g"
```
Comment (3nids) [inline scripts/docker-build.sh:67]:
```suggestion
  grep "ch\/opengis\/qfield" -l -r ${SOURCE_DIR}/src/ | xargs sed -i "s|ch/opengis/qfield|ch/opengis/${APP_PACKAGE_NAME}/g"
```
slightly better readability

File: src/service/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#  PRIVATE $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:QT_QML_DEBUG>)
+
+if (ANDROID)
+    target_link_libraries(qfield_service PRIVATE
+      android
+      log
+    )
+endif()
+
+target_link_libraries(qfield_service PRIVATE
```
Comment (3nids) [inline src/service/CMakeLists.txt:17]: what happens here if not on android?
the android check should be made one level up, no?

File: src/service/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  qfield_core
+  ${QGIS_CORE_LIBRARY}
+)
+
+set_target_properties(qfield PROPERTIES AUTORCC TRUE)
```
Comment (3nids) [inline src/service/CMakeLists.txt:22]:
```suggestion
set_target_properties(qfield_service PROPERTIES AUTORCC TRUE)
```

**Pull request closed without merging**